Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -191,7 +191,7 @@ class SyncRepository {
     private func getChannelIds(completion: @escaping ([ChannelId]) -> Void) {
         database.backgroundReadOnlyContext.perform {
             let request = ChannelDTO.allChannelsFetchRequest
-            request.fetchLimit = 1000
+            request.fetchLimit = 250
```

Comment:
When looking at the code it seems like, if there are 500 channels, then we would only get 250 of them and the other 250 is never synced, only when opening it manually with a channel controller.

Another interesting thing with syncing is that it pauses all the other API requests until sync repo is done. Not sure why it is a requirement.

I guess what I am saying is that sync repo should need a review and the current change is nice!

Reply:
Didn't we say we will use 100? https://getstream.slack.com/archives/CRB9P6L03/p1712674377370179?thread_ts=1712674016.356819&cid=CRB9P6L03

Also, shouldn't we also cleanup/delete the rest of the channels?

Reply:
I did not notice that comment sorry. I also had Auto merge enabled, so the PR got merged when @laevandus approved 😄

Reply:
The cleanup maybe better to make it part of a different PR, since it needs a strategy where we will do this